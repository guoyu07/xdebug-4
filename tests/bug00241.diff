033+       string(99) "array ('GLOBALS' => ..., 'argv' => array (0 => '/Users/johann/src/xdebug/tests/bug00241.php'), ...)"
033-       string(%d) "array ('GLOBALS' => ..., '_ENV' => array ('%s' => '%s', '%s' => '%s', ...), ...)"
049+       string(0) ""
049-       string(4) "NULL"