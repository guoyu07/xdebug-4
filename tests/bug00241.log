
---- EXPECTED OUTPUT
array(5) {
  [0]=>
  array(4) {
    ["function"]=>
    string(6) "{main}"
    ["file"]=>
    string(%d) "%sbug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(0) {
    }
  }
  [1]=>
  array(4) {
    ["function"]=>
    string(13) "error_handler"
    ["file"]=>
    string(%d) "%sbug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(5) {
      ["errno"]=>
      string(1) "8"
      ["string"]=>
      string(22) "'Undefined index: FOO'"
      ["file"]=>
      string(%d) "'%sbug00241.php'"
      ["line"]=>
      string(2) "32"
      ["context"]=>
      string(%d) "array ('GLOBALS' => ..., '_ENV' => array ('%s' => '%s', '%s' => '%s', ...), ...)"
    }
  }
  [2]=>
  array(5) {
    ["function"]=>
    string(8) "newError"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(%d) "%sbug00241.php"
    ["line"]=>
    int(4)
    ["params"]=>
    array(1) {
      ["errno"]=>
      string(4) "NULL"
    }
  }
  [3]=>
  array(5) {
    ["function"]=>
    string(11) "__construct"
    ["class"]=>
    string(11) "Error_Entry"
    ["file"]=>
    string(%d) "%sbug00241.php"
    ["line"]=>
    int(11)
    ["params"]=>
    array(2) {
      ["base"]=>
      string(5) "FALSE"
      ["errno"]=>
      string(5) "FALSE"
    }
  }
  [4]=>
  array(5) {
    ["function"]=>
    string(5) "getBT"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(%d) "%sbug00241.php"
    ["line"]=>
    int(26)
    ["params"]=>
    array(0) {
    }
  }
}
The End
---- ACTUAL OUTPUT
array(5) {
  [0]=>
  array(4) {
    ["function"]=>
    string(6) "{main}"
    ["file"]=>
    string(42) "/Users/dsp/dev/c/xdebug/tests/bug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(0) {
    }
  }
  [1]=>
  array(4) {
    ["function"]=>
    string(13) "error_handler"
    ["file"]=>
    string(42) "/Users/dsp/dev/c/xdebug/tests/bug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(5) {
      ["errno"]=>
      string(1) "8"
      ["string"]=>
      string(22) "'Undefined index: FOO'"
      ["file"]=>
      string(44) "'/Users/dsp/dev/c/xdebug/tests/bug00241.php'"
      ["line"]=>
      string(2) "32"
      ["context"]=>
      string(98) "array ('GLOBALS' => ..., 'argv' => array (0 => '/Users/dsp/dev/c/xdebug/tests/bug00241.php'), ...)"
    }
  }
  [2]=>
  array(5) {
    ["function"]=>
    string(8) "newError"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(42) "/Users/dsp/dev/c/xdebug/tests/bug00241.php"
    ["line"]=>
    int(4)
    ["params"]=>
    array(1) {
      ["errno"]=>
      string(0) ""
    }
  }
  [3]=>
  array(5) {
    ["function"]=>
    string(11) "__construct"
    ["class"]=>
    string(11) "Error_Entry"
    ["file"]=>
    string(42) "/Users/dsp/dev/c/xdebug/tests/bug00241.php"
    ["line"]=>
    int(11)
    ["params"]=>
    array(2) {
      ["base"]=>
      string(5) "FALSE"
      ["errno"]=>
      string(5) "FALSE"
    }
  }
  [4]=>
  array(5) {
    ["function"]=>
    string(5) "getBT"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(42) "/Users/dsp/dev/c/xdebug/tests/bug00241.php"
    ["line"]=>
    int(26)
    ["params"]=>
    array(0) {
    }
  }
}
The End
---- FAILED
