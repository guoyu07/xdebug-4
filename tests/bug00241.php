<?php
function error_handler($errno, $string, $file, $line, $context)
{
	$entry = Error_Class::newError();
}

class Error_Class
{
	public static function newError($errno = false)
	{
		return new Error_Entry(false, $errno);
	}

	public static function getBT()
	{
		$tmp = xdebug_get_function_stack();
		var_dump($tmp);
	}

}

class Error_Entry
{
	public function __construct($base, $errno)
	{
		Error_Class::getBT();
	}
}

set_error_handler('error_handler');

$tmp = explode('/', trim($_SERVER['FOO'], '/'));
echo "The End\n";
?>
