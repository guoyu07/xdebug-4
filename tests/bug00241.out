array(5) {
  [0]=>
  array(4) {
    ["function"]=>
    string(6) "{main}"
    ["file"]=>
    string(43) "/Users/johann/src/xdebug/tests/bug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(0) {
    }
  }
  [1]=>
  array(4) {
    ["function"]=>
    string(13) "error_handler"
    ["file"]=>
    string(43) "/Users/johann/src/xdebug/tests/bug00241.php"
    ["line"]=>
    int(0)
    ["params"]=>
    array(5) {
      ["errno"]=>
      string(1) "8"
      ["string"]=>
      string(22) "'Undefined index: FOO'"
      ["file"]=>
      string(45) "'/Users/johann/src/xdebug/tests/bug00241.php'"
      ["line"]=>
      string(2) "32"
      ["context"]=>
      string(99) "array ('GLOBALS' => ..., 'argv' => array (0 => '/Users/johann/src/xdebug/tests/bug00241.php'), ...)"
    }
  }
  [2]=>
  array(5) {
    ["function"]=>
    string(8) "newError"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(43) "/Users/johann/src/xdebug/tests/bug00241.php"
    ["line"]=>
    int(4)
    ["params"]=>
    array(1) {
      ["errno"]=>
      string(0) ""
    }
  }
  [3]=>
  array(5) {
    ["function"]=>
    string(11) "__construct"
    ["class"]=>
    string(11) "Error_Entry"
    ["file"]=>
    string(43) "/Users/johann/src/xdebug/tests/bug00241.php"
    ["line"]=>
    int(11)
    ["params"]=>
    array(2) {
      ["base"]=>
      string(5) "FALSE"
      ["errno"]=>
      string(5) "FALSE"
    }
  }
  [4]=>
  array(5) {
    ["function"]=>
    string(5) "getBT"
    ["class"]=>
    string(11) "Error_Class"
    ["file"]=>
    string(43) "/Users/johann/src/xdebug/tests/bug00241.php"
    ["line"]=>
    int(26)
    ["params"]=>
    array(0) {
    }
  }
}
The End